modules/standard/Random.chpl

```diff
@@ -89,12 +128,87 @@ record SeedGenerators {
   }
 };
 
-const SeedGenerator: SeedGenerators;
+
+// CHPLDOC FEEDBACK: If easy, I'd suggest either deprecating the 
+// :arg <type> <name>: form or else switching the order to 
+// :arg <name> <type>: as it's an easy trap to simply type the same
+// argument as in the function's signature.
+
+
+// CHPLDOC FIXME: int(64) prints out as int(64)(64)
+
+// CHPLDOC FIXME: if the first line below is shifted one character to the
+// left, it ends up being rendered like a method
+
+// CHPLDOC FIXME: if the first line below is shifted one character to the
+// right, it ends up causing a warning (promoted to error) in the .rst
+// file.  It'd be preferable to have it declare the issue in terms of
+// the .chpl line numbers.
+
+/*
+  Fill an array of real(64), imag(64), or complex(128) elements with
+  pseudorandom values in parallel using a new
+  :chpl:class:`RandomStream` created for the lifetime of this routine.
+  The first `arr.size` values from the stream will be assigned to the
+  array's elements in row-major order for `real` and `imag` elements.
+  For complex elements, consecutive pairs of random numbers are
+  assigned to the real and imaginary components, respectively.  The
+  parallelization strategy is determined by the array.
+
+  :arg arr: The array to be filled, where T is real(64), imag(64), or complex(128).
+  :type arr: [] T
+
+  :arg seed: The seed to use for the PRNG.  Defaults to :chpl:const:`SeedGenerator`.currentTime.
```

Comment:
Oh, I missed the singleton issue. In that case, I might use:

```
:chpl:proc:`SeedGenerator.currentTime <SeedGenerators.currentTime>`
```

Then the link text says what you mean, but it links to the current time proc on SeedGenerators.